SPECS/annobin/annobin.spec

```diff
@@ -176,79 +262,143 @@ touch doc/annobin.info
-# If building on RHEL7, enable the next option as the .attach_to_group assembler pseudo op is not available in the assembler.
+BUILD_FLAGS="-fplugin=%{_tmppath}/tmp_annobin.so"
+
+OPTS="$(rpm --undefine=_annotated_build --eval '%build_cflags %build_ldflags')"
+
+# If building on systems with an assembler that does not support the
+# .attach_to_group pseudo op (eg RHEL-7) then enable the next line.
 # BUILD_FLAGS="$BUILD_FLAGS -fplugin-arg-tmp_annobin-no-attach"
-make -C gcc-plugin CXXFLAGS="%{optflags} $BUILD_FLAGS"
+
+make -C gcc-plugin CXXFLAGS="$OPTS $BUILD_FLAGS"
 rm %{_tmppath}/tmp_annobin.so
 
+cp clang-plugin/annobin-for-clang.so %{_tmppath}/tmp_annobin.so
+# To enable verbose more in the plugin append the following: ANNOBIN="verbose"
+make -C clang-plugin clean all CLANG_TARGET_OPTIONS="$CLANG_TARGET_OPTIONS $BUILD_FLAGS" 
+
+cp llvm-plugin/annobin-for-llvm.so %{_tmppath}/tmp_annobin.so
+# To enable verbose more in the plugin append the following: ANNOBIN_VERBOSE="true"            
+make -C llvm-plugin clean all CLANG_TARGET_OPTIONS="$CLANG_TARGET_OPTIONS $BUILD_FLAGS"
+
 #---------------------------------------------------------------------------------
 
 %install
-%make_install
-%{__rm} -f %{buildroot}%{_infodir}/dir
+
+# PLUGIN_INSTALL_DIR is used by the Clang and LLVM makefiles...
+%make_install PLUGIN_INSTALL_DIR=%{buildroot}/%{llvm_plugin_dir}
+
+# Move the clang plugin to a seperate directory.
+mkdir -p %{buildroot}/%{clang_plugin_dir}
+mv %{buildroot}/%{llvm_plugin_dir}/annobin-for-clang.so %{buildroot}/%{clang_plugin_dir}
+
+# Record the version of gcc that built this plugin.
+# Note - we cannot just store %%{gcc_vr} as sometimes the gcc rpm version changes
+# without the NVR being altered.  See BZ #2030671 for more discussion on this.
+mkdir -p                             %{buildroot}/%{ANNOBIN_GCC_PLUGIN_DIR}
+echo "%{gcc_vr}" > %{buildroot}/%{ANNOBIN_GCC_PLUGIN_DIR}/%{aver}
+
+# Also install a copy of the sources into the build tree.
+mkdir -p                            %{buildroot}%{annobin_source_dir}
+cp %{_sourcedir}/%{annobin_sources} %{buildroot}%{annobin_source_dir}/latest-annobin.tar.xz
+
+rm -f %{buildroot}%{_infodir}/dir
 
 #---------------------------------------------------------------------------------
 
-%if %{with tests}
 %check
-# Change the following line to "make check || :" on RHEL7 or if you need to see the
-# test suite logs in order to diagnose a test failure.
-make check
+# The first "make check" is run with "|| :" so that we can capture any logs
+# from failed tests.  The second "make check" is there so that the build
+# will fail if any of the tests fail.
+make check || :
 if [ -f tests/test-suite.log ]; then
     cat tests/test-suite.log
 fi
-%endif
+# If necessary use uuencode to preserve test binaries here.  For example:
+#   uuencode tests/tmp_atexit/atexit.strip atexit.strip
+
+make check
 
 #---------------------------------------------------------------------------------
 
-%files
```

Comment:
Why did we remove the default `annobin` package? I think I'm a bit confused about what's happening or just missed a line (it's a long change).

Reply:
Hmm, I imported it from Fedora. My understanding is that the default package is a subset of subpackages it produces so it does not have any files on its own

Reply:
Looks like Fedora just doesn't have the default package (for the record: a `%files` section must exist for an RPM to be produced). Our version of `annobin` seems to have only docs and the files, which are moved to `plugin-gcc` in this change. Please double-check, if any of the other packages had a dependency on `annobin`. Otherwise, I think this is fine.

Reply:
We do not have any packages that depend on it. I am only use subpackage `annocheck` from this package